### ISSUE TITLE
Build on Raspberry Pi 3B running a 64-bit kernel fails 

### ISSUE THREAD

**Issue by DougieLawson** [SDL-Hercules-390/hyperion#244]
I've got a Raspberry Pi running Raspbian Buster with the 64-bit 4.19.69-v8+ kernel. 

I pulled the zip from at https://github.com/SDL-Hercules-390/hyperion/commit/18343fb953af79f63db1c0d9a244629cc54b628c unzipped it to a shared directory. Ran `./util/bldlvchk`, `./autogen.sh`, then `CFLAGS="-mcpu=native" ./configure` and `make -j6` which fails with 

```
...
  CC       bootstrap.o
make[2]: *** No rule to make target 'crypto/lib/aarch64/libcrypto64.a', needed by 'hercules'.  Stop.
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/shared/hercules-4.2'
make[1]: *** [Makefile:2448: all-recursive] Error 1
make[1]: Leaving directory '/shared/hercules-4.2'
make: *** [Makefile:1796: all] Error 2
```

**Comment by wably** [comment]
It looks like you did not build the required external packages.  In your Hercules files, please refer to readme/README.EXTPKGS.  Then restart your build.

I can successfully build SDL 4.2 on a Pi 3B and I've attached below my notes of the steps I follow to do it.  I'd recommend that you review it first, before starting the process; it's not long.

Regards,
Bob

[Building SDL Hyperion 4.2.txt](https://github.com/SDL-Hercules-390/hyperion/files/3585880/Building.SDL.Hyperion.4.2.txt)

**Comment by Peter-J-Jansen** [comment]
This week I successfully built SDL Hercules on a Raspberry Pi (RPI) 4 with 4GB of memory, using the unofficial Ubuntu 18.04.3, 64 bit, of James A. Chambers (please see [https://jamesachambers.com/raspberry-pi-4-ubuntu-server-desktop-18-04-3-image-unofficial/](url)). I've kept verbatim installation notes, which might be useful, also for building Hercules other RPI models. I include them down below.

My next steps on this RPI 4 build will be to do it as a pure server, only using SSH to access the Hercules console on the RPI, and tn3270's to access the mainframe screens. So far I've seen Hercules reporting MIPS maxrates as high as 76 MIPS ...

Cheers,
Peter

[SDL Hercules on Raspberry Pi 4 Installation Notes - Ubuntu 18.04.3 Xfce](https://www.dropbox.com/s/075k6x9byejc773/SDL%20Hercules%20on%20Raspberry%20Pi%204%20Installation%20Notes%20-%20Ubuntu%2018.04.3%20Xfce.txt?dl=0)

**Comment by DougieLawson** [comment]
Telling me to read https://github.com/SDL-Hercules-390/hyperion/blob/master/readme/README.EXTPKG.md is like Arthur Dent's experience in HHGTTG.

> “But the plans were on display…”
> “On display? I eventually had to go down to the cellar to find them.”
> “That’s the display department.”
> “With a flashlight.”
> “Ah, well, the lights had probably gone.”
> “So had the stairs.”
> “But look, you found the notice, didn’t you?”
> “Yes,” said Arthur, “yes I did. It was on display in the bottom of a  locked filing cabinet stuck in a disused lavatory with a sign on the door saying ‘Beware of the Leopard.”

How about adding something to the main README.md in the root of the github repo?

It still fails with
```
make[2]: *** No rule to make target '../hercules-libs/lib/aarch64/libcrypto64.a', needed by 'hercules'.  Stop.
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/shared/hercules-4.2'
make[1]: *** [Makefile:2448: all-recursive] Error 1
make[1]: Leaving directory '/shared/hercules-4.2'
make: *** [Makefile:1796: all] Error 2
```

**Comment by wably** [comment]
I agree that reading through README.EXTPKGS is not the easiest thing.  It has been a source of frustration for a while; that's why I kept notes.

I know that you either did not follow the procedure, or that you did not follow it correctly because of the mention of 'aarch64' in the message output in your latest reply.  This means you did not edit the .ini file to set the cpu type to 'arm'.

I urge you to follow the steps I outlined in the 'Building SDL Hyperion 4.2' attachment I posted earlier.  It will get you through the external packages problem much more quickly than trying to read through the README file and re-interpret it for Linux use.

Regards,
Bob
